src/Common/src/System/Globalization/IdnMapping.Unix.cs

```diff
@@ -11,44 +11,39 @@ private unsafe string GetAsciiCore(char* unicode, int count)
             uint flags = Flags;
             CheckInvalidIdnCharacters(unicode, count, flags, nameof(unicode));
 
-            const int StackAllocThreshold = 512;
-            if (count < StackAllocThreshold)
-            {
-                char* output = stackalloc char[count];
-                return GetAsciiCore(unicode, count, flags, output, count, reattempt: true);
-            }
-            else
+            // Each unicode character is represented by 2 ASCII chars
+            // and a "xn--" prefix of length 4
```

Comment:
Nit: this wording makes it sound like there's a length-4 prefix for each Unicode character.